homeassistant/components/abode/__init__.py

```diff
@@ -89,7 +89,7 @@ def __init__(self, abode, polling):
 
         self.abode = abode
         self.polling = polling
-        self.devices = []
+        self.entities = []
```

Comment:
We're not allowed to store entities in a shared central container. Entities are a platform concern. It's ok to store a reference to the entity eg the entity_id.

We can use the dispatch helper to signal the correct entity method from the service handler.

Reply:
Just to make sure I'm understanding this, you're saying I can remove `self.entities = []` here since I'll be storing a reference to the `entity_id` in `self.hass.data[DOMAIN]`?

Reply:
On a second thought, can I just store a list containing `entity_id` within something like `self.entity_ids = []`?

That way I can just iterate over the list within the `capture_image` and `trigger_quick_action` methods. All it appears those methods are doing is comparing the `entity_id` that is being sent from the service call to what actually exists.

I would then just have to make sure I remove `entity_id` when `async_unload_entry` is called.

Reply:
It's good to keep it stored here. Make it a set of entity_ids.